flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/optimization/Solver.scala

```diff
@@ -40,6 +44,32 @@ abstract class Solver extends Serializable with WithParameters {
     data: DataSet[LabeledVector],
     initialWeights: Option[DataSet[WeightVector]]): DataSet[WeightVector]
 
+  /** Creates initial weights vector, creating a DataSet with a WeightVector element
+    *
+    * @param initialWeights An Option that may contain an initial set of weights
+    * @param data The data for which we optimize the weights
+    * @return A DataSet containing a single WeightVector element
+    */
+  def createInitialWeightsDS(initialWeights: Option[DataSet[WeightVector]],
+                             data: DataSet[LabeledVector]):  DataSet[WeightVector] = {
+    // TODO: Faster way to do this?
+    val dimensionsDS = data.map(_.vector.size).reduce((a, b) => b)
+
+    initialWeights match {
+      // Ensure provided weight vector is a DenseVector
+      case Some(wvDS) =>
+        wvDS.map { wv => {
+          val denseWeights = wv.weights match {
+            case dv: DenseVector => dv
+            case sv: SparseVector => sv.toDenseVector
+          }
+          WeightVector(denseWeights, wv.intercept)
```

Comment:
Not sure what you mean here, is it `WeightVector(denseWeights, wv.intercept)` that is off?